core/domain/activity_jobs_one_off.py

```diff
@@ -628,3 +629,111 @@ def reduce(key, values):
+            'id_string_format': 'skill-%s-%s'
+        },
+        'ExplorationRightsSnapshotMetadataModel': {
+            'parent_model_class': exp_models.ExplorationRightsModel,
+            'commit_log_model_class': exp_models.ExplorationCommitLogEntryModel,
+            'id_string_format': 'rights-%s-%s'
+        }
+    }
+    # This list consists of the snapshot models whose parent model has the
+    # post commit status.
+    MODELS_WITH_COMMIT_STATUS = [
+        'ExplorationSnapshotMetadataModel',
+        'ExplorationRightsSnapshotMetadataModel'
+    ]
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        job_class = AddMissingCommitLogsJob
+        return job_class.SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS
+
+    @staticmethod
+    def map(snapshot_model):
+        job_class = AddMissingCommitLogsJob
+        class_name = snapshot_model.__class__.__name__
+        model_id, version = snapshot_model.id.rsplit('-', 1)
+        model_properties = job_class.MODEL_NAMES_TO_PROPERTIES[class_name]
+        commit_log_id = (
+            model_properties['id_string_format'] % (model_id, version))
+        commit_log_model_class = (
+            model_properties['commit_log_model_class'].get_by_id(
+                commit_log_id))
+        commit_logs_should_exist = True
+
+        if class_name in job_class.MODELS_WITH_COMMIT_STATUS:
+            parent_model = (
+                model_properties['parent_model_class'].get_by_id(model_id))
+        if class_name == 'ExplorationRightsSnapshotMetadataModel':
+            if snapshot_model.commit_type in ['create', 'delete']:
+                commit_logs_should_exist = False
+
+        if commit_log_model_class is None and commit_logs_should_exist:
+            # Public is the default value for post_commit_status for the
+            # question and skill models. For the exp model, it is assigned
+            # from the rights model.
+            commit_log_model = model_properties['commit_log_model_class'](
+                id=commit_log_id,
+                user_id=snapshot_model.committer_id,
+                commit_type=snapshot_model.commit_type,
+                commit_message=snapshot_model.commit_message,
+                commit_cmds=snapshot_model.commit_cmds,
+                post_commit_status=constants.ACTIVITY_STATUS_PUBLIC,
+                version=int(version)
+            )
+            if class_name == 'SkillSnapshotMetadataModel':
+                commit_log_model.skill_id = model_id
+            elif class_name == 'QuestionSnapshotMetadataModel':
+                commit_log_model.question_id = model_id
+            elif class_name == 'ExplorationRightsSnapshotMetadataModel':
+                commit_log_model.exploration_id = model_id
+                commit_log_model.post_commit_status = parent_model.status
+            elif class_name == 'ExplorationSnapshotMetadataModel':
+                commit_log_model.exploration_id = model_id
```

Comment:
```suggestion
            setattr(commit_log_model, model_properties['id_field'], model_id)
```

and `model_properties['id_field']` would be for example `'skill_id'`.

Reply:
Done.